### PR TITLE
Various improvements to C API bindings

### PR DESCRIPTION
**Pull request by d-e-s-o**
Please see individual commits for descriptions.

### CODE REVIEW

File: src/c_api.rs [hunk 1]
```diff
@@ -686,8 +684,6 @@ pub enum blazesym_sym_type {
 #[repr(C)]
 #[allow(non_camel_case_types, unused)]
 pub enum blazesym_faf_type {
-    /// Invalid type
-    BLAZESYM_FAF_T_INVALID,
```
Comment (anakryiko) [inline src/c_api.rs]: I'm a bit undecided on this. Judging by a lot of previous C-side experiences with kernel UAPIs, having 0 match to something invalid and requiring users to actively set non-zero values is proving pretty valuable in UAPI design.

File: src/c_api.rs [hunk 2]
```diff
@@ -732,9 +728,6 @@ unsafe fn convert_find_addr_features(
                     blazesym_sym_type::BLAZESYM_SYM_T_VAR => {
                         FindAddrFeature::SymbolType(SymbolType::Variable)
                     }
-                    _ => {
-                        panic!("Invalid symbol type");
```
[the remark this answers is not in the document]
Comment (danielocfb) [reply src/c_api.rs:733]: It's undefined behavior, see https://github.com/libbpf/blazesym/issues/84. That's what it was before, that's what it is now. There is no change.

File: src/c_api.rs (hunk 1 above)
Comment (danielocfb) [reply src/c_api.rs]: I agree that our variants should have non-zero values. I was planning on getting there when we address https://github.com/libbpf/blazesym/issues/84.
However, having an "invalid" variant makes no sense. All it forces us to do is have a two staged validation process: first we need to check whether we have a valid Rust `enum` and then we need to check whether it's invalid. It has no meaning, so this variant should not exist.

Comment (anakryiko) [reply src/c_api.rs]: alright

**Comment by danielocfb** [comment]
I rebased, fixed conflicts, and merged the change. Thanks for the review!

**Pull request closed without merging**